pkg/storaged/btrfs/subvolume.jsx

```diff
@@ -0,0 +1,117 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from "cockpit";
+import React from "react";
+
+import { CardBody } from "@patternfly/react-core/dist/esm/components/Card/index.js";
+import { DescriptionList } from "@patternfly/react-core/dist/esm/components/DescriptionList/index.js";
+
+import { StorageCard, StorageDescription, new_card, new_page } from "../pages.jsx";
+import { StorageUsageBar } from "../storage-controls.jsx";
+import { decode_filename, get_fstab_config_with_client } from "../utils.js";
+import { btrfs_usage } from "./utils.jsx";
+import { mounting_dialog } from "../filesystem/mounting-dialog.jsx";
+import client from "../client.js";
+
+const _ = cockpit.gettext;
+
+function subvolume_mount_info(volume, subvol) {
+    const fsys = client.blocks_fsys[volume.path];
+    const block = client.blocks[volume.path];
+    const mount_points = fsys.MountPoints.map(decode_filename);
+
+    const mount_point = get_fstab_config_with_client(client, block, false, subvol)[1];
+    const is_mounted = mount_points.indexOf(mount_point) >= 0;
+    return [mount_point, is_mounted];
+}
+
+function subvolume_unmount(volume, subvol) {
+    const block = client.blocks[volume.path];
+    mounting_dialog(client, block, "unmount", null, subvol);
+}
+
+function subvolume_mount(volume, subvol) {
+    const block = client.blocks[volume.path];
+    mounting_dialog(client, block, "mount", null, subvol);
+}
+
+export function make_btrfs_subvolume_page(parent, volume, subvol) {
+    const [mount_point, is_mounted] = subvolume_mount_info(volume, subvol);
+    console.log(volume, subvol);
+    let location = "";
+    const use = btrfs_usage(client, volume);
+    const actions = [];
+
+    if (mount_point) {
+        location = mount_point;
+        if (is_mounted) {
+            actions.push({
+                title: _("Unmount"),
+                action: () => subvolume_unmount(volume, subvol),
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19690-20231205-110913-7b3be2c9-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)